tests/RobotFramework/tests/cumulocity/telemetry/child_device_telemetry.robot

```diff
@@ -70,6 +70,27 @@ Child devices support sending inventory data via c8y topic
     Should Be Equal    ${mo["custom"]["fragment"]}    yes
 
 
+Child devices support sending inventory data via tedge topic with type
+    Execute Command    tedge mqtt pub "te/device/${CHILD_SN}///data/device_OS" '{"family":"Debian","version":11,"complex":[1,"2",3],"object":{"foo":"bar"}}'
+    Cumulocity.Set Device    ${CHILD_SN}
+    ${mo}=    Device Should Have Fragments    device_OS
+    Should Be Equal    ${mo["device_OS"]["family"]}    Debian
+    Should Be Equal As Integers    ${mo["device_OS"]["version"]}    11
+
+    Should Be Equal As Integers    ${mo["device_OS"]["complex"][0]}    1
+    Should Be Equal As Strings    ${mo["device_OS"]["complex"][1]}    2
+    Should Be Equal As Integers    ${mo["device_OS"]["complex"][2]}    3
+    Should Be Equal    ${mo["device_OS"]["object"]["foo"]}    bar
+
+
+Child devices supports sending inventory data via tedge topic to root fragments
+    Execute Command    tedge mqtt pub "te/device/${CHILD_SN}///data/" '{"subtype":"LinuxDeviceA","type":"ShouldBeIgnored"}'
+    Cumulocity.Set Device    ${CHILD_SN}
+    ${mo}=    Device Should Have Fragments    subtype
+    Should Be Equal    ${mo["subtype"]}    LinuxDeviceA
+    Should Be Equal    ${mo["type"]}    thin-edge.io
```

Comment:
Nice to have tests along the specs ♥

This one makes me understand why a `type` fragment is an issue as already defined by `tedge config get device.type`. But, don't we have to address with a broader view these conflicts with legacy features (`inventory.json` and `tedge config`)?

Reply:
The `inventory.json` contents can be parsed and any direct key-value pairs can be published to the `meta/key` topic, with the `value` as the payload. The same can be extended to `tedge config` as well, where the `value` would be just be the plain value instead of a JSON object, as you suggested.

Reply:
We could go the layered approach and first parse the inventory.json file and use that as the base model, and then apply any changes from the MQTT topics. The inventory.json is good for being able to hardcode static information into a build, however we could also just phase the file out (behind a feature flag or something).

Reply:
@reubenmiller So, no need to re-publish the contents of the file as `twin` messages to the broker? Without that, the other components relying on the broker for the inventory information will never get that "base model", right? Unless they all parse that file.